### PR TITLE
BUG: fix empty sjoin

### LINKED ISSUE #731: Value error using sjoin with pandas v0.23
I use the `sjoin` function to add the region name (polygons) to every point within the region. Some points are not in any region, therefore I filter these points and buffer them step by step. So the points layer without intersection becomes smaller and smaller. If there is only one row left I get the following error in pandas v0.23 which I did not get before (pandas < v0.23). Using geopandas v0.3.0.

My call:
```python
new = gpd.sjoin(rest_points, polygons, how='left', op='intersects')
```

Error message:
```python
ValueError: You are trying to merge on object and int64 columns.
If you wish to proceed you should use pd.concat
```
class: `GeoDataFrame`
method: `merge(self, *args, **kwargs)`
line: `result = DataFrame.merge(self, *args, **kwargs)`

I do not understand the error and why it happens only with the last point (last row) and only with the newest pandas version. I had a look at "What's New" but could not find anything.

Full message:
```python
  File "virtualenv/lib/python3.5/site-packages/geopandas/tools/sjoin.py", line 140,
    in sjoin suffixes=('_%s' % lsuffix, '_%s' % rsuffix))
  File "virtualenv/lib/python3.5/site-packages/geopandas/geodataframe.py", line 418,
     in merge result = DataFrame.merge(self, *args, **kwargs)
  File "virtualenv/lib/python3.5/site-packages/pandas/core/frame.py", line 6379,
     in merge copy=copy, indicator=indicator, validate=validate)
  File "virtualenv/lib/python3.5/site-packages/pandas/core/reshape/merge.py", line 60,
     in mergevalidate=validate)
  File "virtualenv/lib/python3.5/site-packages/pandas/core/reshape/merge.py", line 554,
     in __init__self._maybe_coerce_merge_keys()
  File "virtualenv/lib/python3.5/site-packages/pandas/core/reshape/merge.py", line 980,
        in _maybe_coerce_merge_keys
    raise ValueError(msg)
ValueError: You are trying to merge on object and int64 columns.
If you wish to proceed you should use pd.concat
```

### PR DESCRIPTION
**Pull request by ljwolf**
Closes #731

This makes sure that the key dtype in empty spatial joins is `float`, addressing but #731.

### CODE REVIEW

File: geopandas/tools/tests/test_sjoin.py [hunk 1]
```diff
@@ -117,6 +117,18 @@ def test_left(self, op, dfs):
 
         assert_frame_equal(res, exp)
 
+        # Check empty joins on left
```
Comment (jorisvandenbossche) [inline geopandas/tools/tests/test_sjoin.py]: Can you maybe make this just a separate test? (`test_empty_join` or something)

Comment (ljwolf) [reply geopandas/tools/tests/test_sjoin.py]: sure, yes

File: geopandas/tools/tests/test_sjoin.py [hunk 2]
```diff
@@ -117,6 +117,23 @@ def test_left(self, op, dfs):
 
         assert_frame_equal(res, exp)
 
+    def test_empty_join(self):
+            # Check empty joins on left
```
Comment (jorisvandenbossche) [inline geopandas/tools/tests/test_sjoin.py]: minor nit, but the indentation seems a bit too much here?

Comment (ljwolf) [reply geopandas/tools/tests/test_sjoin.py]: yep, no, was not intended. thanks for catching this.

**Pull request merged**